stores/types.go

```diff
@@ -35,8 +35,41 @@ type (
 	balance        big.Int
 	unsigned64     uint64 // used for storing large uint64 values in sqlite
 	secretKey      []byte
+	setting        string
 )
 
+// GormDataType implements gorm.GormDataTypeInterface.
+func (setting) GormDataType() string {
+	return "string"
+}
+
+// String implements fmt.Stringer to prevent "s3authentication" settings from
+// getting leaked.
+func (s setting) String() string {
+	if strings.Contains(string(s), "v4Keypairs") {
```

Comment:
This works but it's a little crude maybe? Should we try and unmarshal the setting to `JSON` and loop the keys if that works? Allowing us to only obfuscate only the field instead of the entire setting value.

Reply:
I considered that but unmarshalling json is going to be a lot slower for a setting that contains only 1 field anyway and that is the `v4Keypairs` field. We can reconsider if we ever end up extending that setting with fields we don't want to hide.

Reply:
Ok, I'll approve the PR but I would've done the unmarshalling, we don't really need to care about it being slower I think and missing information in the logs might end up being annoying when we have to debug an issue.